src/lang.rs

```diff
@@ -110,9 +110,14 @@ pub enum InlineOperator {
     Where {
         expr: Expr,
     },
+    Limit {
+        /// The count for the limit is pretty loosely typed at this point, the next phase will
+        /// check the value to see if it's sane or provide a default if no number was given.
+        count: Option<f64>,
```

Comment:
I was trying to avoid doing any semantic checks in the parser and just accept a "number".  Since nom provided a `double` parser, I figured it was easiest to use that and then validate the value in the outer level.  Similarly, I used Option so that the resulting syntax tree would capture what the user provided.